catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/CswFilterFactory.java

```diff
@@ -835,4 +837,38 @@ private BinarySpatialOpType createBinarySpatialOpType(PropertyNameType propertyN
         return binarySpatialOpType;
     }
 
+    //default implementation just assume the first argument is a parameter name and the rest are values
+    public FilterType buildPropertyIsEqualTo(String functionName, List<Object> arguments,
+            Object literal) {
+        int propertyCount = 1;
+        List<JAXBElement<?>> expressions = new ArrayList<>();
+        for (int i = 0; i < arguments.size(); i++) {
+            if (i < propertyCount) {
+                expressions.add(createPropertyNameType(Arrays.asList(new Object[] {
```

Comment:
I don't disagree here are some options:
A. (What I did) I can make an assumption that covers most cases so that this doesn't need to be implemented for most functions we add.
B. I can throw an exception to force this to be updated for every new function we implement if we remember to check if CSW federation works.
C. I can just treat them all as literals because a String from a property name is going to be treated (Mostly) the same as a string from a literal (when these functions are parsed from CQL the filter object has them as literals anyway so we will have this issue no matter what).
D. Rework everything to pass down more information from the FunctionFactory to the filter delegates about which values are PropertyNames and which are Values.